apis/apps/v1alpha1/opsrequest_webhook.go

```diff
@@ -416,14 +416,74 @@ func (r *OpsRequest) validateHorizontalScaling(_ context.Context, _ client.Clien
 	if len(horizontalScalingList) == 0 {
 		return notEmptyError("spec.horizontalScaling")
 	}
-
+	for _, v := range r.Spec.HorizontalScalingList {
+		if v.Operator != HScaleDeleteOP {
+			continue
+		}
+		for _, comSpec := range cluster.Spec.ComponentSpecs {
+			if err := r.validateHorizontalScalingSpec(v, comSpec, comSpec.Name, false); err != nil {
+				return err
+			}
+		}
+		for _, shardingSpec := range cluster.Spec.ShardingSpecs {
+			if err := r.validateHorizontalScalingSpec(v, shardingSpec.Template, shardingSpec.Name, true); err != nil {
+				return err
+			}
+		}
+	}
 	compOpsList := make([]ComponentOps, len(horizontalScalingList))
 	for i, v := range horizontalScalingList {
 		compOpsList[i] = v.ComponentOps
 	}
 	return r.checkComponentExistence(cluster, compOpsList)
 }
 
+func (r *OpsRequest) validateHorizontalScalingSpec(hScale HorizontalScaling, compSpec ClusterComponentSpec, componentName string, isSharding bool) error {
+	if hScale.ComponentName != componentName {
+		return nil
+	}
+	if hScale.AutoSyncReplicas {
+		if isSharding {
+			return fmt.Errorf("can not auto-sync replicas when the specified component is a sharding component")
```

Comment:
why?

Reply:
shardingSpecs[*].template.replicas affects all shards, usually when offline instances, it is necessary to keep the replicas unchanged. Unless explicitly specified to scale out the replicas